test/framework/toolchain.py

```diff
@@ -755,7 +755,7 @@ def test_old_new_iccifort(self):
         self.modtool.purge()
 
         tc = self.get_toolchain('goolfc', version='1.3.12')
-        tc.options.update({'openmp': True})
+        tc.set_options({'openmp': True})
```

Comment:
@akesandgren We've been using -fopenmp for a while now with the Intel compilers, actually si ce they deprecated -openmp and recommended using -qopenmp instead. The problem with the latter is that it only works with sufficiently recent versions of the Intel compilers. Also, I don't see them deprecating -fopenmp, since that would break compatibility with GCC...

So, I propose we stick to -fopenmp, and revisit when necessary?